scipy/stats/stats.py

```diff
@@ -2829,7 +2829,12 @@ def median_absolute_deviation(x, axis=0, center=np.median, scale=1.4826,
 
     # Consistent with `np.var` and `np.std`.
     if not x.size:
-        return np.nan
+        nan_shape = [item for i, item in enumerate(x.shape) if i != axis]
```

Comment:
`axis` might be negative, and if it is, this line does work correctly.

Reply:
Oh. OK. In this implementation, just returning no.nan if axis number is negative or bigger than the shape dimension. I will try to fix it.

Reply:
oops, good catch @WarrenWeckesser. `axis = axis % x.ndim` should fix that

Reply:
PTAL #11747

Reply:
@rgommers, that's not quite right either, because that will accept something crazy like `axis=999`.  Any value not in `range(-x.ndim, x.ndim)` should raise an exception.

Reply:
Specifically, it should raise `np.AxisError`.